### PR TITLE
Allow smartstack status when not verbose

### LINKED ISSUE #767: Smarstack status is not shown when not verbose unless nerve_ns has a service definition
# Issue description

When the nerve_ns for all services points to a service that does not have a definition, such as when we have `main_sfo2`,`main_iad1` but no `main`, smartstack status is not shown on `paasta status`.

# Issue examples

Without the issue, normal service with a `main:` definition:
```
cluster: norcal-prod
instance: main_uswest1
Git sha:    fbdbbc0c (desired)
State:      Running - Desired state: Started
Marathon:   Healthy - up with (4/4) instances. Status: Running
Mesos:      Healthy - (4/4) tasks in the TASK_RUNNING state.
Smartstack: N/A - main_uswest1 is announced in the main namespace.
instance: main
Git sha:    fbdbbc0c (desired)
State:      Running - Desired state: Started
Marathon:   Healthy - up with (4/4) instances. Status: Running
Mesos:      Healthy - (4/4) tasks in the TASK_RUNNING state.
Smartstack:
    sfo12-prod - Healthy - in haproxy with (6/6) total backends UP in this namespace.
    uswest1-prod - Healthy - in haproxy with (6/6) total backends UP in this namespace.
instance: main_sfo
Git sha:    fbdbbc0c (desired)
State:      Running - Desired state: Started
Marathon:   Healthy - up with (4/4) instances. Status: Running
Mesos:      Healthy - (4/4) tasks in the TASK_RUNNING state.
Smartstack: N/A - main_sfo is announced in the main namespace.
```

With the issue, service with no `main:` definition, but with `nerve_ns: main`:
```
cluster: norcal-prod
instance: main_uswest1
Git sha:    fbdbbc0c (desired)
State:      Running - Desired state: Started
Marathon:   Healthy - up with (4/4) instances. Status: Running
Mesos:      Healthy - (4/4) tasks in the TASK_RUNNING state.
Smartstack: N/A - main_uswest1 is announced in the main namespace.
instance: main_sfo
Git sha:    fbdbbc0c (desired)
State:      Running - Desired state: Started
Marathon:   Healthy - up with (4/4) instances. Status: Running
Mesos:      Healthy - (4/4) tasks in the TASK_RUNNING state.
Smartstack: N/A - main_sfo is announced in the main namespace.
```

### PR DESCRIPTION
**Pull request by danielhoherd**
Closes #767

### CODE REVIEW

**Comment by solarkennedy** [comment]
Looks like it needs a little TLC. You can reproduce the test failures by running "tox".

**Comment by solarkennedy** [comment]
Why did you close it? I think it is on the right path? Want to pair up for a bit to get a jumpstart into how to iterate on this?

**Comment by danielhoherd** [comment]
@solarkennedy I accidentally deleted the branch, which closed it. I am still working on it, but I'd love to meet and get some pointers whenever you get a few minutes.

**Comment by danielhoherd** [comment]
I reworked this with most of the logic from `test_status_smartstack_backends_normal`. My reason for doing so is that it's possible to have a single service definition without the `main` namespace, and if so you would want all of the same tests to pass as if it were inside the `main` namespace. This may have been heavy handed though, maybe some of these tests aren't strictly relevant?

File: paasta_tools/marathon_serviceinit.py [hunk 1]
```diff
@@ -267,9 +267,6 @@ def status_smartstack_backends(service, instance, job_config, cluster, tasks, ex
     if instance != nerve_ns:
```
Comment (solarkennedy) [inline paasta_tools/marathon_serviceinit.py]: I would also junk lines 267-269. It isn't N/A anymore if we are outputting stuff

File: tests/test_marathon_serviceinit.py [hunk 2]
```diff
@@ -429,26 +429,74 @@ def test_status_smartstack_backends_normal():
 def test_status_smartstack_backends_different_nerve_ns():
     service = 'my_service'
     instance = 'my_instance'
-    cluster = 'fake_cluster'
     different_ns = 'other_instance'
-    normal_count = 10
-    tasks = mock.Mock()
-    with mock.patch('paasta_tools.marathon_tools.read_namespace_for_service_instance', autospec=True) as read_ns_mock:
-        read_ns_mock.return_value = different_ns
+    service_instance = compose_job_id(service, different_ns)
```
Comment (solarkennedy) [inline tests/test_marathon_serviceinit.py]: I know this is silly, but can you drop this line and make the assertion be:
```
service_instance="my_service.other_namespace"
```

And just hard code it in the test.

The reasoning here is that I want to make it very obvious that the function is called with the other namespace, and not the normal "service_instance"

File: tests/test_marathon_serviceinit.py [hunk 3]
```diff
@@ -429,26 +429,74 @@ def test_status_smartstack_backends_normal():
 def test_status_smartstack_backends_different_nerve_ns():
     service = 'my_service'
     instance = 'my_instance'
-    cluster = 'fake_cluster'
     different_ns = 'other_instance'
```
Comment (solarkennedy) [inline tests/test_marathon_serviceinit.py]: Don't call it "other_instance" just call it "different_ns" to make the test read more obviously.

In this context it isn't really another instance, it is just another nerve namespace.

**Comment by solarkennedy** [comment]
Thanks!

**Pull request merged**